boot/boot.js

```diff
@@ -287,6 +287,136 @@ $tw.utils.decodeURIComponentSafe = function(s) {
 	return v;
 };
 
+/*************
+Code to make the permalink/permaview URLs more readable.
+(These comments are just to help reviewers; they will eventually be removed.)
+***************/
+
+
+/*************
+Helpers for encoding/decoding URIs.  **Note** that these are throw-away, only
+used to temporarily build `charMatch`, `spaceMatch`, `sentenceEnding`, and
+`sentenceTrailing`, all of which will eventually be hard-coded.
+***************/
+
+/*
+The character that will substitute for a space in the URL
+*/
+var SPACE = "_";
+
+/*
+The character added to the end to avoid ending with `.`, `?`, `!` or the like
+*/
+var TRAILER = '_';
```

Comment:
Every character and string definition in this file should use double quotes for consistency. 
We only use single quotes if there is no other chance or to make it easier for translators. Eg: for translation strings to avoid string escaping. But that's basically it.

Reply:
Sorry about that.

I swear this things proliferate on their own!  I'm quite sure that my decades of a different coding style of nothing to do with it.  Nothing at all.  😜   Last I tried, there were so many lint warnings that I wasn't able to identify ones I was adding.